### PR TITLE
MM-12753 & MM-15569 Fix missing posts when getting posts since

### PR DESCRIPTION
**Pull request by saturninoabril**
#### Summary
Fix missing posts when getting posts since.  It's happening when limiting number of posts in two separate statements before union.

This PR is the last item for unread API.  Once merged, I'll submit `unread-posts-api` against master.

#### Ticket Link
Jira ticket
- [MM-12753](https://mattermost.atlassian.net/browse/MM-12753)
- [MM-15569](https://mattermost.atlassian.net/browse/MM-15569)

### CODE REVIEW

File: store/sqlstore/post_store.go [hunk 1]
```diff
@@ -371,7 +371,7 @@ func (s *SqlPostStore) InvalidateLastPostTimeCache(channelId string) {
 func (s *SqlPostStore) GetEtag(channelId string, allowFromCache bool) store.StoreChannel {
 	return store.Do(func(result *store.StoreResult) {
 		if allowFromCache {
-			if cacheItem, ok := s.lastPostTimeCache.Get(channelId); ok && cacheItem.(int64) > 0 {
+			if cacheItem, ok := s.lastPostTimeCache.Get(channelId); ok {
```
Comment (hmhealey) [inline store/sqlstore/post_store.go]: What's the reason for this change? I don't understand why that extra check was there in the first place, but I'm not sure why it would we removed

File: store/sqlstore/post_store.go [hunk 2]
```diff
@@ -597,8 +597,8 @@ func (s *SqlPostStore) GetPostsSince(channelId string, time int64, limit int, al
 			    WHERE
 			        UpdateAt > :Time
 			            AND ChannelId = :ChannelId
-			    LIMIT :Limit) temp_tab))
-			ORDER BY CreateAt DESC`,
+			    LIMIT 1000) temp_tab))
+			ORDER BY CreateAt ASC LIMIT :Limit`,
```
Comment (hmhealey) [inline store/sqlstore/post_store.go]: Why change the order from descending to ascending?

Comment (saturninoabril) [reply store/sqlstore/post_store.go]: __My original thought is that:__ I changed it 'cause I think the old implementation of `GetPostsSince` does not sound accurate.  The existing limit of 1000 seems to mean return "all" posts from a certain time to new post.  We don't have any issue with it being "all" since our client usually don't ask for a very far "since" and as a result it returns less than 1000 posts.

However with unread API, it only gets limited number of posts around last unread (or a period that the channel was last viewed) and returns a number of post (say 60) from such period (old) to recent post).  If we keep DESC and put limit of 60 on it, it will return 60 posts starting from recent to old, and as a result it will going to have missing messages in between.

However, upon looking closer on the changes I've made, I realized that I should have not touch the GetPostSince and should just stick on GetPostsAfter.  I'll revise the PR...

**Comment by saturninoabril** [comment]
@enahum @hmhealey Sorry for the inconvenience in reviewing this PR.  I messed this PR and the branch it's submitted against to.  I ported the changes and comments to https://github.com/mattermost/mattermost-server/pull/11039.
Closing this PR.
